dsp-role/core/src/main/scala/dsp/role/domain/RoleDomain.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright © 2021 - 2022 Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package dsp.role.domain
+
+import dsp.errors.BadRequestException
+import dsp.valueobjects.Id.RoleId
+import dsp.valueobjects.Id.UserId
+import dsp.valueobjects.Permission
+import dsp.valueobjects.Role._
+import zio.prelude.Validation
+
+case class RoleUser(
+  id: UserId
+)
+
+/**
+ * Role's domain model.
+ *
+ * @param id the role's ID
+ * @param name the role's name
+ * @param description the role's description
+ * @param users the role's users
+ * @param permission the role's permission
+ */
+sealed abstract case class Role private (
+  id: RoleId,
+  name: LangString,
+  description: LangString,
+  users: List[RoleUser], // List[User]
+  permission: Permission
+) extends Ordered[Role] { self =>
+
+  /**
+   * Allows to compare the [[Role]] instances.
+   *
+   * @param that [[Role]] to compare
+   * @return [[Boolean]] value
+   */
+  def compare(that: Role): Int = self.id.iri.toString().compareTo(that.id.iri.toString())
+  //  Boolean = self.id.equals(that.id)
+
+  /**
+   * Updates the role's name.
+   *
+   * @param newValue new role's name to update
+   * @return updated [[Role]]
+   */
+  def updateName(newValue: LangString): Validation[BadRequestException, Role] =
```

Comment:
In opposite to that `r: Role` here maybe the param name could be name/newName, because type doesn't say what it is, but:
-  `def updateName(newValue: LangString)` is very clear the method will update the Name and requires its **new value** in `LangString` type to be passed
- we have the docString which clarify it even more